File: Implementierung/Backend/src/main/java/edu/kit/informatik/dto/mapper/interactions/QualityMapper.java
package edu.kit.informatik.dto.mapper.interactions;

import edu.kit.informatik.dto.mapper.IModelDtoMapper;
import edu.kit.informatik.dto.userdata.interactions.QualityDto;
import edu.kit.informatik.model.userdata.interactions.Quality;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;

@Service
public class QualityMapper implements IModelDtoMapper<Quality, QualityDto> {

    @Override
    public QualityDto modelToDto(Quality quality) {
        return QualityDto.valueOf(quality.toString());
    }

    @Override
    public List<QualityDto> modelToDto(List<Quality> qualities) {
        List<QualityDto> qualityDtos = new ArrayList<>();
        qualities.forEach(quality -> qualityDtos.add(modelToDto(quality)));

        return qualityDtos;
    }

    @Override
    public Quality dtoToModel(QualityDto qualityDto) {
        return Quality.valueOf(qualityDto.toString());
    }

    @Override
    public List<Quality> dtoToModel(List<QualityDto> qualityDtos) {
        List<Quality> qualities = new ArrayList<>();
        qualityDtos.forEach(qualityDto -> qualities.add(dtoToModel(qualityDto)));

        return qualities;
    }
}

File: Implementierung/frontend/paschi/src/dto/mapper/interactions/QualityMapper.ts
import {IModelDtoMapper} from "@/dto/mapper/IModelDtoMapper";
import {Quality} from "@/model/userdata/interactions/Quality";
import {QualityDto} from "@/dto/userdata/interactions/QualityDto";

export class QualityMapper implements IModelDtoMapper<Quality, QualityDto> {

  private static mapper: QualityMapper = new QualityMapper();

  private constructor() {
  }

  static getMapper(): QualityMapper {
    return QualityMapper.mapper;
  }

  dtoToModel(qualityDto: QualityDto): Quality {
    switch (qualityDto) {
      case QualityDto.ONE_STAR:
        return Quality.ONE_STAR;
      case QualityDto.TWO_STAR:
        return Quality.TWO_STAR;
      case QualityDto.THREE_STAR:
        return Quality.THREE_STAR;
      case QualityDto.FOUR_STAR:
        return Quality.FOUR_STAR;
      case QualityDto.FIVE_STAR:
        return Quality.FIVE_STAR;
    }
  }

  modelToDto(quality: Quality): QualityDto {
    switch (quality) {
      case Quality.ONE_STAR:
        return QualityDto.ONE_STAR;
      case Quality.TWO_STAR:
        return QualityDto.TWO_STAR;
      case Quality.THREE_STAR:
        return QualityDto.THREE_STAR;
      case Quality.FOUR_STAR:
        return QualityDto.FOUR_STAR;
      case Quality.FIVE_STAR:
        return QualityDto.FIVE_STAR;
    }
  }
}
